libcloudforensics/providers/kubernetes/mitigation.py

```diff
@@ -30,40 +32,49 @@ def DrainWorkloadNodesFromOtherPods(
         to prevent pods from appearing on the nodes again as it will be marked
         as unschedulable. Defaults to True.
   """
-  nodes = set(pod.GetNode() for pod in workload.GetCoveredPods())
+  nodes = workload.GetCoveredNodes()
   if cordon:
     for node in nodes:
       node.Cordon()
   for node in nodes:
-    node.Drain(lambda p: not workload.IsCoveringPod(p))
+    node.Drain(lambda pod: not workload.IsCoveringPod(pod))
 
 
-def CreateDenyAllNetworkPolicyForWorkload(
+def IsolatePodsWithNetworkPolicy(
     cluster: k8s.K8sCluster,
-    workload: workloads.K8sWorkload) -> netpol.K8sDenyAllNetworkPolicy:
-  """Isolates a workload's pods via a deny all network policy.
+    pods: List[base.K8sPod]) -> Optional[netpol.K8sDenyAllNetworkPolicy]:
+  """Isolates pods via a deny-all NetworkPolicy.
+
+  **Warning:** It is the caller's responsibility to make sure that Kubernetes
+  NetworkPolicy is enabled for their cluster, via their cloud provider's API.
 
   Args:
-    cluster (k8s.K8sCluster): The cluster in which to create the deny
-        all policy, and subsequently patch existing policies
-    workload (workloads.K8sWorkload): The workload in whose namespace the
-        deny all network policy will be created, and whose pods will be tagged
-        to be selected by the deny all network policy.
+    cluster (k8s.K8sCluster): The cluster in which to create the deny-all
+        policy.
+    pods (List[base.K8sPod]): The pods to patch with the labels of the created
+        deny-all NetworkPolicy.
 
   Returns:
-    netpol.K8sDenyAllNetworkPolicy: The deny all network policy that was
-        created to isolate the workload's pods.
+    netpol.K8sDenyAllNetworkPolicy: Optional. The deny-all network policy that
+        was created to isolate the pods. If no pods were supplied, None is
+        returned.
+
+  Raises:
+    ValueError: If the pods are not in the same namespace.
   """
-  # TODO: Check that network policies are enabled
```

Comment:
Is there a reason why we don't check and error if it's not, instead of putting this responsibility on the user?

Reply:
Yes, the intention was to make this `kubernetes` package standalone, and not rely on any other APIs than the Kubernetes API. If we want checks to be integrated, then we can move these functions to the GCP package. The problem I'd see with this is that it may lead to code duplication when we support AWS/Azure clusters.

Reply:
Okay so a solution that may work is making the `K8sCluster` abstract, and have an abstract method that checks if NetworkPolicy is enabled, and have the cloud provider packages subclass it. We can then check if it is enabled in this function.

There is already a class restructuring branch that I have, so I'll add this change there, in a separate PR. I've made an issue for this #400